### DISCUSSION TITLE
Minio - Azure AKS to use Azure Blob Storage

### DISCUSSION THREAD

**Discussion by ankur-jaiswal1981** [minio/minio#15830] [Q&A] [answered]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

## Expected Behavior

Earlier, Minio provided the Azure Gateway Functionality as "interface" to connect the Azure Blob Storage as a PV or PVC. However, after "Azure Gateway" was deprecated there has been no update as how we can now use _"Azure Blob Storage with Minio on _AKS"__ 

## Current Behavior

I tried with GUI mode of Minio on AKS - there has been no option for the _Azure Blob Storage CSI _driver__

![image](https://user-images.githubusercontent.com/91355874/194838378-b4e5ebee-0e4f-451c-8192-81d5fa399c73.png)

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Context

I'm trying to use Minio as an interface to attach the _Azure Blob _Storage__ as a PV and PVC to Azure AKS.

## Your Environment

* Version used (`minio --version`): helm.sh/chart:minio-operator-4.3.7
* Server setup and configuration: Minio Operator and Console installed on Azure AKS. 
* Operating System and version (`uname -a`): AKSUbuntu-1804gen2containerd-2022.08.29

**Comment by vadmeste** [comment]
@ankur-jaiswal1981 can you describe how you deployed MinIO AKS ?

**Comment by ankur-jaiswal1981** [comment]
Yep. Here you go - 
1.   Azure AKS was installed natively via the Azure GUI. 
2.   This Helm chart : helm.sh/chart:minio-operator-4.3.7, was used to instantiate the Minio Operator and Minio Console - both running as a pod in Azure AKS.
3.   Got the JWT using the `Kubectl get `secret``
4.   Port forwarded the console pods running in AKS.
5.   Accessed the GUI by putting the JWT obtained in Step 3.
6.   Once the GUI was accessed - tried to create a "TENANT" - and after this i wasn't able to see the Azure Blob Storage as a CSI for AKS.
![image](https://user-images.githubusercontent.com/91355874/194849064-d7983b61-0810-4ea4-89e7-1328a4d1ff62.png)

**Comment by vadmeste** [comment, marked as answer]
You need to manually install Azure Blob CSI, https://learn.microsoft.com/en-us/azure/aks/azure-blob-csi?tabs=NFS

**Reply by ankur-jaiswal1981** [reply]
Yep this Worked out thanks!

**Comment by ankur-jaiswal1981** [comment]
Could you also let me know : When we create a Tenant 
1. What are [ number of servers ] here : Is it a VM or the Physical Nodes [ Worker Nodes in AKS ]?
2. Number of Drives per Server : Ideally it should be the Blob Storage if am using the AzureBlobCSI then why it's saying the Drives?
3. I am not able to see the architecture diagram which clearly explains the association of these resources [ Servers and Memory across Nodes ]
![image](https://user-images.githubusercontent.com/91355874/196101431-9d293794-1ba0-41b5-93c9-f7975af1ccc3.png)
